CHANGELOG.md

```diff
@@ -1,5 +1,36 @@
 # go-ipfs changelog
 
+### 0.3.4 - unreleased
+
+#### API changes
+
+##### `core/coreunix`'s `Add` → `shell/fsnode`'s `AddFromReader`
+
+We've added an `AddFromReader` function to `shell/fsutils`.  The old
+`Add` from `core/coreunix` is deprecated and will be removed in
+version 0.4.0. To update your existing code, change usage like:
+
+    keyString, err := coreunix.Add(ipfsNode, reader)
+    if err != nil {
+        return err
+    }
+
+to
+
+    fileNode, err := fsutils.AddFromReader(ipfsNode, reader)
+    if err != nil {
+        return err
+    }
+    key, err := fileNode.Key()
+    if err != nil {
+        return err
+    }
+    keyString := key.String()
+
+That's a bit more verbose if all you want is the stringified key, but
+returning a `dag.Node` makes it easier to perform other operations
+like adding the file node to a directory node.
```

Comment:
we're not currently doing descriptions like this. seems very nice and useful for consumers.

the binary's changelog is not the best place. this changelog is just a tiny summary. maybe we can have a different file for things like this.

we should define clearly what interfaces we commit to as stable. in particular:

- core
- shell

come to mind as the main ones. changes in these are potentially large.